### PR TITLE
Add quiz class

### PR DESCRIPTION
**Pull request by kstonekuan**
Part of #11 but still needs extra implementation with other classes

### CODE REVIEW

File: src/main/java/seedu/eduke8/question/QuestionList.java [hunk 1]
```diff
@@ -18,7 +18,7 @@ public QuestionList(ArrayList<QuestionInterface> loadedQuestions) {
 
 
     @Override
-    public ArrayList<QuestionInterface> getChosenNumberOfRandomQuestion(int numberOfQuestionsForQuiz) {
+    public void setQuizQuestions(int numberOfQuestionsForQuiz) {
```
Comment (Zhi-You) [inline src/main/java/seedu/eduke8/question/QuestionList.java:21]: Thanks for this! I think this implementation makes more sense!

File: src/main/java/seedu/eduke8/question/QuestionList.java [hunk 2]
```diff
@@ -70,7 +68,7 @@ public QuestionInterface getNextQuestion() {
     }
 
     @Override
-    public Boolean allQuestionsAnswered() {
+    public boolean areAllQuestionsAnswered() {
```
Comment (Zhi-You) [inline src/main/java/seedu/eduke8/question/QuestionList.java:71]: Oh right! My mistake on the primitive versus object type for the boolean variable! Thanks for helping me edit that!
I think the new naming you have here is much clearer too!

File: src/main/java/seedu/eduke8/question/QuestionListInterface.java [hunk 3]
```diff
@@ -17,6 +15,6 @@
 
     QuestionInterface getNextQuestion();
 
-    Boolean allQuestionsAnswered();
+    boolean areAllQuestionsAnswered();
```
Comment (anqi20) [inline src/main/java/seedu/eduke8/question/QuestionListInterface.java:18]: All these naming makes it easier to understand!

File: src/main/java/seedu/eduke8/quiz/Quiz.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    private TopicInterface topic;
+    private int numberOfQuestions;
+
+    public Quiz(TopicInterface topic, int numberOfQuestions) {
+        this.topic = topic;
+        this.numberOfQuestions = numberOfQuestions;
+    }
+
+    @Override
+    public void startQuiz(UiInterface ui) {
```
Comment (Zhi-You) [inline src/main/java/seedu/eduke8/quiz/Quiz.java:21]: Great work! I think the logic is fine for the quiz implementation!

**Review by Zhi-You — APPROVED** [review]
LGTM! With the quiz logic in place, our working product should be up and running soon!

**Pull request merged**